packages/integrations/image/src/utils/paths.ts

```diff
@@ -5,13 +5,19 @@ import type { TransformOptions } from '../loaders/index.js';
 import { isRemoteImage } from './images.js';
 import { shorthash } from './shorthash.js';
 
+function removeQueryString(src: string) {
+	const index = src.lastIndexOf('?');
+	return index > 0 ? src.substring(0, index) : src;
+}
+
 export function ensureDir(dir: string) {
 	fs.mkdirSync(dir, { recursive: true });
 }
 
 export function propsToFilename({ src, width, height, format }: TransformOptions) {
-	const ext = path.extname(src);
-	let filename = src.replace(ext, '');
+	let filename = removeQueryString(src);
+	const ext = path.extname(filename);
+	filename = filename.replace(ext, '');
```

Comment:
probably unlikely to happen but this replaces the first substring of the extension so if you had `foo.js.bar.js` it would become `foo.bar.js`. Using `lastIndexOf` to find where the extension is located and then substringing up to that point would be more technically correct I think.

Reply:
Great catch, I handled that for stripping the querystring but forgot to move that over to `lastIndexOf + substring` for the extension

fixed in the latest commit :+1: